src/Domain/Donation.php

```diff
@@ -1415,7 +1415,9 @@ public function update(
         $this->setTipGiftAid($tipGiftAid);
         $this->setTbgShouldProcessGiftAid($this->getCampaign()->getCharity()->isTbgClaimingGiftAid());
         $this->setDonorHomePostcode($donorHomePostcode);
-        $this->setDonorName($donorName);
+        if ($donorName) {
```

Comment:
One of the tests in `UpdateHandlesLockExceptionTest` would bail out without a message being dispatched, due to the lack of a first / any donor name, without this condition being included. I think there's probably other test changes that could have addressed the problem, but it seems prudent not to let one version of the property wipe out the other version and potentially already-persisted data in any case.